diff --git a/sysconfig/network-scripts/ifup b/sysconfig/network-scripts/ifup
index d7bd672..56b2d6e 100755
--- a/sysconfig/network-scripts/ifup
+++ b/sysconfig/network-scripts/ifup
@@ -75,7 +75,7 @@ if [ "${VLAN}" = "yes" ] && [ "$ISALIAS" = "no" ]; then
     VID=""
     MATCH='^(eth|hsi|bond)[0-9]+\.[0-9]{1,4}$'
     if [[ "${DEVICE}" =~ $MATCH ]]; then
-	VID=$(echo "${DEVICE}" | LC_ALL=C sed 's/^[a-z0-9]*\.0*//')
+	VID=$(echo "${DEVICE}" | LC_ALL=C sed 's/^[a-z0-9]*\.0*\([0-9]\+\)/\1/')
 	PHYSDEV=${DEVICE%.*}
     fi
     MATCH='^vlan[0-9]{1,4}?'
